scripts/lib/setup-apt-repo

```diff
@@ -23,12 +29,21 @@ done
 # Ensure the directory for LAST_DEPENDENCIES_HASH exists
 mkdir -p /var/lib/zulip
 
-SOURCES_FILE=/etc/apt/sources.list.d/zulip.list
-STAMP_FILE=/etc/apt/sources.list.d/zulip.list.apt-update-in-progress
+SOURCES_FILE=/etc/apt/sources.list.d/$LIST.list
+STAMP_FILE=/etc/apt/sources.list.d/$LIST.list.apt-update-in-progress
 
-ZULIP_SCRIPTS="$(dirname "$(dirname "$0")")"
-DEPENDENCIES_HASH=$(sha1sum "$ZULIP_SCRIPTS/setup/"*.asc "$0")
-DEPENDENCIES_HASH_FILE="/var/lib/zulip/setup-repositories-state"
+ZULIP_SCRIPTS="$(cd "$(dirname "$(dirname "$0")")" && pwd)"
+LIST_PATH="$ZULIP_SCRIPTS/setup/apt-repos/$LIST"
+if ! [ -d "$LIST_PATH" ]; then
+    echo "Not a valid value for --list: '$LIST'"
+    echo ""
+    echo "Valid values are:"
+    ls -1 "$ZULIP_SCRIPTS/setup/apt-repos/"
```

Comment:
Maybe we want this error message to talk a bit about how to add new platforms?  

I guess one concept would be a link to new little section at the bottom of https://zulip.readthedocs.io/en/latest/subsystems/dependencies.html that's a checklist for adding new versions, that perhaps @ligmitz can write as a follow-up to this PR.

Reply:
This section isn't platform-dependent -- it's for if someone tries to say `scripts/lib/setup-apt-repo --list ksplice-uptrace` instead of `scripts/lib/setup-apt-repo --list ksplice`.

I'll add a little more wording down below for the "unsupported release" check.